run_build.sh

```diff
@@ -115,12 +115,19 @@ function check_daos_linkage() {
 
 function install_python_deps() {
     echo "Installing python build dependencies"
-    cmd="python3 -m pip install --upgrade pip"
+    cmd="python3 -m pip install --user --upgrade pip"
```

Comment:
--user not needed?

Reply:
I think maybe your comment was for an older commit. Anyway, --user is not necessary, but passing --user silences a verbose warning from pip

Reply:
@ravalsam I must have read the git diff wrong the first time. I did add --user here to be explicit. This is what happens without --user:
```
python3 -m pip install --upgrade pip
Defaulting to user installation because normal site-packages is not writeable
```
So, it uses --user anyway because we don't have permission for the global installation. What do you think?

Reply:
oh, I did read that wrong my bad. yes --user is required so was not sure if that was working without --user.

Reply:
No worries. Looks like we both read it wrong :)